### PR TITLE
Some bits of randao

### PR DESCRIPTION
**Pull request by yglukhov**
- Added `validator_keygen`
- Added randao to block proposal

### CODE REVIEW

File: beacon_chain/validator_pool.nim [hunk 1]
```diff
@@ -64,3 +63,11 @@ proc signAttestation*(v: AttachedValidator,
     # send RPC
     discard
 
+proc randaoReveal*(v: AttachedValidator, commitment: Eth2Digest): Future[Eth2Digest] {.async.} =
+  if v.kind == inProcess:
+    result = v.randaoSecret.reveal(commitment)
```
Comment (zah) [inline beacon_chain/validator_pool.nim]: I'm adding `await sleepAsync(1)` to all validator methods to simulate a certain level of asynchronicity.

Comment (cheatfate) [reply beacon_chain/validator_pool.nim]: @zah 1ms is not enough to simulate async, use at least 50ms.

File: beacon_chain/randao.nim [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import spec/digest
+
+type Randao* = object
+  seed: Eth2Digest
+
+const MaxRandaoLevels = 10000 # TODO: This number is arbitrary
```
Comment (arnetheduck) [inline beacon_chain/randao.nim]: What's this number for? in the spec, the number of levels that should be applied increases under certain conditions as time progresses for a particular validator, what's the intended use here?

File: beacon_chain/randao.nim [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  if commitment == r.seed:
+    raise newException(Exception, "Randao: cannot reveal for seed")
+  result = r.seed
+  for i in 0 .. MaxRandaoLevels:
+    let h = eth2hash(result.data)
+    if h == commitment:
+      return
+    result = h
+
+  raise newException(Exception, "Randao: commitment not found")
```
Comment (arnetheduck) [inline beacon_chain/randao.nim]: not really unexpected/exceptional - in terms of code, we should probably move pure logic like this outside of any exception raising glue

File: beacon_chain/validator_pool.nim (hunk 1 above)
Comment (arnetheduck) [reply beacon_chain/validator_pool.nim]: a bit of a fallacy to enforce any particular ordering in async, be it late or early..

File: beacon_chain/randao.nim (hunk 3 above)
Comment (arnetheduck) [reply beacon_chain/randao.nim]: actually, it already exists in `spec` - I'll provide a patch

**Pull request merged**